test/phase-2/scWETHv2.t.sol

```diff
@@ -0,0 +1,1513 @@
+            abi.encodeWithSelector(ISwapRouter.swap0x.selector, swapData, wstEthAmount)
+        );
+        assertGe(weth.balanceOf(address(this)), expectedWethAmount, "weth not received");
+        assertEq(wstEth.balanceOf(address(this)), 0, "wstEth not transferred out");
+    }
+
+    function test_invest_FloatBalanceTooSmall(uint256 amount) public {
+        _setUp(BLOCK_BEFORE_EULER_EXPLOIT);
+
+        amount = bound(amount, boundMinimum, 15000 ether);
+        _depositToVault(address(this), amount);
+
+        uint256 investAmount = amount - minimumFloatAmount + 1;
+
+        (scWETHv2.SupplyBorrowParam[] memory supplyBorrowParams,, uint256 totalFlashLoanAmount) =
+            _getInvestParams(investAmount, aaveV3AllocationPercent, eulerAllocationPercent, compoundAllocationPercent);
+
+        // deposit into strategy
+        vm.startPrank(keeper);
+        vm.expectRevert(
+            abi.encodeWithSelector(FloatBalanceTooSmall.selector, minimumFloatAmount - 1, minimumFloatAmount)
+        );
+        vault.investAndHarvest(
+            investAmount, supplyBorrowParams, _getSwapDefaultData(investAmount + totalFlashLoanAmount, 0)
+        );
+    }
+
+    function test_invest_TooMuch(uint256 amount) public {
+        _setUp(BLOCK_BEFORE_EULER_EXPLOIT);
+
+        amount = bound(amount, boundMinimum, 15000 ether);
+        _depositToVault(address(this), amount);
+
+        uint256 investAmount = amount + 1;
+
+        (scWETHv2.SupplyBorrowParam[] memory supplyBorrowParams,, uint256 totalFlashLoanAmount) =
+            _getInvestParams(investAmount, aaveV3AllocationPercent, eulerAllocationPercent, compoundAllocationPercent);
+
+        // deposit into strategy
+        vm.startPrank(keeper);
+        vm.expectRevert(InsufficientDepositBalance.selector);
+        vault.investAndHarvest(
+            investAmount, supplyBorrowParams, _getSwapDefaultData(investAmount + totalFlashLoanAmount, 0)
+        );
+    }
+
+    function test_deposit_eth(uint256 amount) public {
+        _setUp(BLOCK_AFTER_EULER_EXPLOIT);
+
+        amount = bound(amount, boundMinimum, 1e21);
+        vm.deal(address(this), amount);
+
+        assertEq(weth.balanceOf(address(this)), 0);
+        assertEq(address(this).balance, amount);
+
+        vault.deposit{value: amount}(address(this));
+
+        assertEq(address(this).balance, 0, "eth not transferred from user");
+        assertEq(vault.balanceOf(address(this)), amount, "shares not minted");
+        assertEq(weth.balanceOf(address(vault)), amount, "weth not transferred to vault");
+
+        vm.expectRevert("ZERO_SHARES");
+        vault.deposit{value: 0}(address(this));
+    }
+
+    function test_maxLtv() public {
+        _setUp(BLOCK_BEFORE_EULER_EXPLOIT);
+
+        assertEq(aaveV3Adapter.getMaxLtv(), 0.9e18, "aaveV3 Max Ltv Error");
+        assertEq(eulerAdapter.getMaxLtv(), 0.7565e18, "euler Max Ltv Error");
+        assertEq(compoundV3Adapter.getMaxLtv(), 0.9e18, "compoundV3 Max Ltv Error");
+    }
```

Comment:
these also should be moved from this file, make a note to move them later when scUSDC and this branch are merged into main